code/src/test/kotlin/com/expediagroup/sdk/core/authentication/bearer/BearerAuthenticationManagerTest.kt

```diff
@@ -0,0 +1,552 @@
+        verify(exactly = 1) { requestExecutor.execute(any()) }
+    }
+
+    /**
+     * Test isTokenAboutToExpire when token is not about to expire.
+     */
+    @Test
+    fun `isTokenAboutToExpire should return false when token is valid`() {
+        // Arrange
+        val expiresIn: Long = 3600L
+        var available: Int? = null
+        val response = Response.builder()
+            .body(
+                ResponseBody.create(
+                    """{ "access_token": "accessToken", "expires_in": $expiresIn }""".toByteArray().inputStream().also {
+                        available = it.available()
+                    },
+                    CommonMediaTypes.APPLICATION_FORM_URLENCODED,
+                    available!!.toLong()
+                )
+            )
+            .status(Status.ACCEPTED)
+            .protocol(Protocol.HTTP_1_1)
+            .message("Accepted")
+            .request(Request.builder().url("http://localhost").method(Method.POST).build())
+            .build()
+
+        every { requestExecutor.execute(any()) } returns response
+
+        // Act
+        authenticationManager.authenticate()
+        val isAboutToExpire = authenticationManager.isTokenAboutToExpire()
+
+        // Assert
+        assertFalse(isAboutToExpire)
+    }
+
+    /**
+     * Test isTokenAboutToExpire when token is about to expire.
+     */
+    @Test
+    fun `isTokenAboutToExpire should return true when token is about to expire`() {
+        // Arrange
+        val expiresIn: Long = 1L
+        var available: Int? = null
+        val response = Response.builder()
+            .body(
+                ResponseBody.create(
+                    """{ "access_token": "first_token", "expires_in": $expiresIn }""".toByteArray().inputStream().also {
+                        available = it.available()
+                    },
+                    CommonMediaTypes.APPLICATION_FORM_URLENCODED,
+                    available!!.toLong()
+                )
+            )
+            .status(Status.ACCEPTED)
+            .protocol(Protocol.HTTP_1_1)
+            .message("Accepted")
+            .request(Request.builder().url("http://localhost").method(Method.POST).build())
+            .build()
+
+        every { requestExecutor.execute(any()) } returns response
+
+        // Authenticate to store the token
+        authenticationManager.authenticate()
+
+        // Create a spy of the authentication manager to manipulate isTokenAboutToExpire
+        val spyManager = spyk(authenticationManager)
+
+        // Mock isTokenAboutToExpire to return true
+        every { spyManager.isTokenAboutToExpire() } returns true
+
+        // Act
+        val isAboutToExpire = spyManager.isTokenAboutToExpire()
+
+        // Assert
+        assertTrue(isAboutToExpire)
+        verify(exactly = 1) { spyManager.isTokenAboutToExpire() }
+    }
+
+    /**
+     * Test clearing authentication.
+     */
+    @Test
+    fun `clearAuthentication should reset the token`() {
```

Comment:
```suggestion
    fun `should handle token clearance`() {
```

Can we follow the same pattern for the reset of test cases here?

Reply:
done.